rules/type-declaration/tests/Rector/FunctionLike/ParamTypeDeclarationRector/Fixture/dataprovider_array.php.inc

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Rector\TypeDeclaration\Tests\Rector\FunctionLike\ParamTypeDeclarationRector\Fixture;
+
+final class DemoArrayStringTest
+{
+    public function provideThings(): array {
+        return [
+            [ 'I am a string' ],
+        ];
+    }
+
+    /**
+     * @dataProvider provideThings
+     */
+    public function testGetFromId( $id ): void {}
+}
```

Comment:
What should happen here? 

We use `Skip*` naming convention for classes/files that should be skipped.
Or there must be other half test fixture file for of what is expected.

```php
input
-----
expected
```

Same for file bellow

Reply:
So, this could either.
- do the same as in the yield test, meaning no type is added to the parameter 
- add a type but the correct type (not array).

Seemingly either would be possible and count as a "fix".

Reply:
What is a correct type to be added? Seem like an array from `provideThings()`?

Reply:
Well, provideThings indeed returns an array.
But the first parameter that will be received by methods using the data provider will be a `string`

https://getrector.org/demo/f19f03fb-0731-4f6b-939f-086ebf0757b9 is a simple example of rector noticing this and setting the phpdoc @return for the provider to be correct.
Adding a second parameter there in the dataProvider seems to do something odd though.
https://getrector.org/demo/b0d16cc1-3dc9-4f3b-b3f1-6e7e9884d1a5

I'm going to dig into the code a little and see if I can figure out how this is working currently (and label the tests some more (and maybe write some more))
Any pointers appreciated!